src/main/java/org/ow2/proactive/catalog/util/GrantHelper.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive.catalog.util;
+
+import static org.ow2.proactive.catalog.util.AccessType.noAccess;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.ow2.proactive.catalog.dto.BucketGrantMetadata;
+import org.ow2.proactive.catalog.dto.CatalogObjectGrantMetadata;
+import org.ow2.proactive.catalog.dto.GrantMetadata;
+import org.ow2.proactive.catalog.repository.entity.BucketGrantEntity;
+import org.ow2.proactive.catalog.repository.entity.CatalogObjectGrantEntity;
+import org.ow2.proactive.catalog.service.BucketService;
+import org.ow2.proactive.catalog.service.OwnerGroupStringHelper;
+import org.ow2.proactive.catalog.service.model.AuthenticatedUser;
+
+
+public class GrantHelper {
+    private static final String GROUP_GRANTEE_TYPE = "group";
+
+    private static final String USER_GRANTEE_TYPE = "user";
+
+    // bucket owner is in the format of "GROUP:XXXXX"
+    private static final int BUCKET_OWNER_PREFIX_LENGTH = OwnerGroupStringHelper.GROUP_PREFIX.length();
+
+    public static boolean isPublicBucket(String bucketOwner) {
+        return BucketService.DEFAULT_BUCKET_OWNER.equals(bucketOwner);
+    }
+
+    /**
+     * remove the prefix of the bucket owner to get the bucket belongs to which group
+     *
+     * @param bucketOwner bucket owner (complete format, "GROUP:XXXXX")
+     * @return the group name that the bucket belongs to
+     */
+    public static String extractBucketOwnerGroup(String bucketOwner) {
+        return bucketOwner.substring(BUCKET_OWNER_PREFIX_LENGTH);
+    }
+
+    public static BucketGrantMetadata ownerGroupGrant(String bucketOwner, String bucketName) {
+        return new BucketGrantMetadata("group", "", bucketOwner.substring(6), "admin", 5, 0, bucketName);
```

Comment:
if "group:" or "group" is defined somewhere as a constant, it would be cleaner to use

substring(GROUP_PREFIX.length())